src/script/cpp_api/s_security.cpp

```diff
@@ -503,6 +504,133 @@ bool ScriptApiSecurity::safeLoadFile(lua_State *L, const char *path, const char
 	return result;
 }
 
+std::string inline _get_mod_name_from_debug(lua_Debug *info)
+{
+	std::string mod_name;
+	std::string src = info->source;
+	std::string BIN_DIR = std::string(DIR_DELIM) + "bin" + DIR_DELIM;
+	std::string MOD_DIR = BIN_DIR + ".." + DIR_DELIM + "mods" + DIR_DELIM;
+
+	size_t found = src.find(MOD_DIR);
+	if (found != std::string::npos) {
+		// Check the path whether be faked.
+		if (src.find(BIN_DIR) == found) {
```

Comment:
this needs to be describe exactly in which situations the path is "fake" and the additional check is necessary
again, likely to be insecure

Reply:
The mod code subdirectory does not allow the existence of the "bin" directory.